src/core/js/views/buttonsView.js

```diff
@@ -96,28 +92,29 @@ define(function() {
                 var buttonText = this.model.get('_buttons')["_" + changedAttribute].buttonText;
 
                 switch (changedAttribute) {
-                    case "showCorrectAnswer": case "hideCorrectAnswer":
+                    case "submit":
+                    case "showCorrectAnswer":
+                    case "hideCorrectAnswer":
     				    //make model answer button inaccessible but enabled for visual users
     				    //	due to inability to represent selected incorrect/correct answers to a screen reader, may need revisiting
-                        this.$('.buttons-action').a11y_cntrl(false).html(buttonText).attr('aria-label', ariaLabel);
+                        this.$('.buttons-action').a11y_cntrl_enabled(false).html(buttonText).attr('aria-label', ariaLabel);
```

Comment:
So.. Tom and I have had a look together. There are a whole series of issues here.

* updateButtons is a "click event handler" it was really intended to switch the button states from one mode to another on the buttons click. you are using it as part of your refresh functionality to maintain the correct state for the changed attributes.. so it probably shouldn't be used for that at all and the multiple calls are causing havock with states
* updateButtons is being called twice from the questionModel.refresh function. once in the function and then again from the questionView event handler for the ```question:refrseh``` event
* we don't have a state for "incorrect + don't show model answer button" but we should, instead it reverts to "correct" (as this doesn't show the "show model answer" button) even though the question was incorrectly answered
* the assessment doesn't have a handle to the questionView so instead it calls refresh on the questionModel, the questionModel then triggers a refresh event to the listening questionViews... this would work fine if the questionView wasn't then changing the questionModel. It's a horrid call stack, the refresh on the questionModel should make sure the questionModel state is consistent then trigger for the view to update. period.
* you were right about the buttons-action button not being made enabled properly when the state changes, but i don't think changing the current functions is going to help, especially with the above outstanding issues

i'm sure there are a few other bits i've missed... as a result going to have to remove my +1 from the assessment pr because i don't want to complicate an already bad situation and that shouldn't be merged as it needs the questionModel.refresh function.